packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -2268,7 +2268,7 @@ void mult_A_B_newmatrix(
     Cimport = Bimport;
     Ccolmap = Bview.colMap;
     // Bcol2Ccol is trivial
-    Kokkos::parallel_for(range_type(0,Bview.colMap->getNodeNumElements()),KOKKOS_LAMBDA(const size_t i) {
+    Kokkos::parallel_for("Tpetra::mult_A_B_newmatrix::Bcol2Ccol_fill",range_type(0,Bview.colMap->getNodeNumElements()),KOKKOS_LAMBDA(const size_t i) {
         Bcol2Ccol(i) = Teuchos::as<LO>(i);
```

Comment:
`Teuchos::as` is not a device function.  Also, in a debug build (`Teuchos_ENABLE_DEBUG:BOOL=ON`), it may throw an exception on overflow.